sway/tree/root.c

```diff
@@ -29,13 +31,51 @@ struct sway_root *root_create(void) {
 		sway_log(SWAY_ERROR, "Unable to allocate sway_root");
 		return NULL;
 	}
+
+	struct wlr_scene *root_scene = wlr_scene_create();
+	if (!root_scene) {
+		sway_log(SWAY_ERROR, "Unable to allocate root scene node");
+		free(root);
+		return NULL;
+	}
+
 	node_init(&root->node, N_ROOT, root);
-	root->output_layout = wlr_output_layout_create();
-	wl_list_init(&root->all_outputs);
+	root->root_scene = root_scene;
+
+	bool failed = false;
+	root->staging = alloc_scene_tree(&root_scene->tree, &failed);
+	root->layer_tree = alloc_scene_tree(&root_scene->tree, &failed);
```

Comment:
This logic exists ~identically in `output.c`, can we export a function out of `output.c`/`root.c` to do this, and use it here/there? Maybe `output.c` makes more sense because it'd be close to the destruction logic.

Reply:
These are not the same, the root handles more layers then the output.

Reply:
Ah yes, e.g. `fullscreen_global`. That's unfortunate, a standard `__attribute__((packed))` would really have come in handy here.